lib/Dialect/TTNN/IR/TTNNOpsAttrs.cpp

```diff
@@ -68,6 +65,51 @@ bool TTNNLayoutAttr::hasInterleavedDRAMTensorMemoryLayout() const {
          (getMemLayout().getValue() == TensorMemoryLayout::Interleaved);
 }
 
+// Calculate the logical shape of the shard.
```

Comment:
Follow comment style which is used throughout the file and add as much context as needed (`getElementSizeBytes` as example). Same for other functions.